docs/source/design.rst

```diff
@@ -1,7 +1,151 @@
-.. _internals:
+.. _design:
 
-*********************
-Ibis design internals
-*********************
+Design
+======
 
-More to come here.
+Primary Goals
+*************
```

Comment:
I think this should be '----' otherwise you have too many levels of nesting (compared to other sections)

Reply:
Yep, done